src/main/java/net/helpscout/api/utils/ParamsUtils.java

```diff
@@ -0,0 +1,84 @@
+package net.helpscout.api.utils;
+
+import lombok.SneakyThrows;
+
+import java.net.URLEncoder;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * @Author: ivan
+ * Date: 28.04.16
+ * Time: 0:29
+ */
+public class ParamsUtils {
+
+    @SneakyThrows
+    public static String setParams(String url, Map<String, String> params) {
+
+        if (params != null && params.size() > 0) {
+            StringBuilder sb = new StringBuilder();
+            sb.append(url);
+            for (String key : params.keySet()) {
+                appendParamSeparator(sb);
+                String encodedParameter = URLEncoder.encode(params.get(key), "UTF-8");
+                sb.append(key).append("=").append(encodedParameter);
+            }
+            return sb.toString();
+        }
+        return url;
+    }
+
+    public static String setFields(String url, List<String> fields) {
+
+        if (fields != null && fields.size() > 0) {
+            StringBuilder sb = new StringBuilder();
+            sb.append(url);
+            appendParamSeparator(sb);
+            sb.append("fields=");
+
+            String sep = "";
+            for (String field : fields) {
+                sb.append(sep).append(field);
+                sep = ",";
+            }
+            url = sb.toString();
+        }
+        return url;
+    }
+
+    public static Map<String, String> getCustomerSearchParams(String email, String firstName, String lastName, Integer page) {
+
+        Map<String, String> params = new HashMap<String, String>();
+        if (isNotNullAndNotEmpty(email)) {
```

Comment:
I'd argue that it will produce 2 methods instead of one. Still we have pass names of parameters in 4 calls. In addition, there is one integer parameter, which should be checked in caller method and then converted to String to pass to this helper method. It will move validation logic from one place to separate. I would agree if there will be several methods with such logic, in this case, helper method will concentrate all validation logic at one place. From my point of view, they are already in one place and well readable at current implementation.